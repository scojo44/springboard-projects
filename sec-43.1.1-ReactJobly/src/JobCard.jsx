import React from 'react'
import {Link} from 'react-router-dom';
import './JobCard.css'

export default function JobCard({job}) {
  const {id, title, salary, equity, companyHandle} = job;

  return (
    <li className="JobCard">
      <h3>{title}</h3>
      <p>Salary: <span className="detail-value">{salary}</span></p>
      <p>Equity: <span className="detail-value">{equity}</span></p>
    </li>
  )
}
