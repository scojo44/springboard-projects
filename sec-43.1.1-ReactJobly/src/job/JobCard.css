.JobCard {
  display: flex;
  background-color: rgba(0,0,0,30%);
  border-radius: .5rem;
  margin: .75rem;
  padding: .5rem;
  list-style-type: none;
}
.JobCard div:first-child {
  flex: 1;
}
.JobCard div:last-child {
  flex: 0;
  align-self: center;
}
.JobCard .link {
  display: flex;
}
.JobCard div {
  flex: 1;
}
.JobCard h3 {
  margin: 0 0 .5rem 0;
}
.JobCard p {
  margin: 0;
}
.JobCard img {
  min-width: 5rem;
  min-height: 5rem;
}
.JobCard a:link,
.JobCard a:visited {
  color: white;
  text-decoration: none;
}
.JobCard button {
  color: white;
  background-color: indigo;
  font-weight: bold;
  margin-right: 2em;
  padding: .5rem .5rem;
  border-color: indigo;
  border-radius: .4rem;
  width: 6em;
}
.JobCard button:disabled {
  color: lightgray;
  background-color: darkslateblue;
  border-color: darkslateblue;
}