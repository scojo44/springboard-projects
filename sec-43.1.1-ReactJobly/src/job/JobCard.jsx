import React, {useContext} from 'react'
import UserContext from '../UserContext'
import './JobCard.css'

export default function JobCard({job}) {
  const {appliedJobIDs, applyToJob} = useContext(UserContext);
  const {id, title, salary, equity, companyHandle} = job;

  return (
    <li className="JobCard">
      <div>
        <h3>{title}</h3>
        <p>Salary: <span className="detail-value">{salary}</span></p>
        <p>Equity: <span className="detail-value">{equity}</span></p>
      </div>
      <div>
        <button type="button" onClick={() => applyToJob(id)} disabled={appliedJobIDs.has(id)}>
          {appliedJobIDs.has(id)? 'Applied':'Apply'}
        </button>
      </div>
    </li>
  );
}
