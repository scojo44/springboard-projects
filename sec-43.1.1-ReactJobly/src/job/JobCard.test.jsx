import {render} from '@testing-library/react'
import {MemoryRouter} from 'react-router-dom'
import JobCard from './JobCard';

const TEST_JOB = {
  id: 1,
  title: 'Test Engineer',
  salary: 200000,
  equity: .05,
  companyHandle: 'test-corp',
};

describe('JobCard Tests', () => {
  it('Renders without crashing', () => {
    render(
      <MemoryRouter>
        <JobCard job={TEST_JOB} />
      </MemoryRouter>
    );
  });

  it('Matches snapshot', () => {
    const {asFragment} = render(
      <MemoryRouter>
        <JobCard job={TEST_JOB} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('Shows some details for a job', async () => {
    const {getByText} = render(
      <MemoryRouter>
        <JobCard job={TEST_JOB} />
      </MemoryRouter>
    );

    expect(getByText(TEST_JOB.title)).toBeInTheDocument();
    expect(getByText('200000')).toBeInTheDocument();
    expect(getByText('0.05')).toBeInTheDocument();
  });
});
