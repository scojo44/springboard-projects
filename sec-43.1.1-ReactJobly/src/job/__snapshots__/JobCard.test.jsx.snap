// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JobCard Tests > Matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="JobCard"
  >
    <h3>
      Test Engineer
    </h3>
    <p>
      Salary: 
      <span
        class="detail-value"
      >
        200000
      </span>
    </p>
    <p>
      Equity: 
      <span
        class="detail-value"
      >
        0.05
      </span>
    </p>
  </li>
</DocumentFragment>
`;
